Included Docker image generation

hugoh opened the pull request:
Here's a proposed patch to solve #42.

To make it work: do a search and replace of hugohh/tmo-monitor into your know Docker Hub username/repo, and add the corresponding DOCKERHUB_USERNAME and DOCKERHUB_TOKEN as repository secrets.

highvolt-dev commented:
Thanks for prepping the PR, @hugoh

Seems that now that this would be integrated into the repo that it wouldn't need to git clone in the `Dockerfile` and could instead use Docker `add` along with a `.dockerignore` file

hugoh commented:
I did remove the `git clone` in favor of a`COPY`.

Good point on `.dockerignore`.

highvolt-dev commented:
> I did remove the `git clone` in favor of a`COPY`.

Oops, was reviewing from my phone and saw `
LABEL org.opencontainers.image.source` referencing the repo URL

hugoh commented:
Integrated your comment about `.dockerignore`.

highvolt-dev commented:
@hugoh I plan to incorporate this PR this weekend - probably tomorrow. Sorry for the wait!

hugoh commented:
I updated the patch for the refactoring (and fixed a bug), but FYI, the `pip3 install` doesn't seem to install the modules correctly:

```
Traceback (most recent call last):
  File "/usr/local/bin/tmo-monitor.py", line 6, in <module>
    from tmo_monitor.gateway.model import GatewayModel
ModuleNotFoundError: No module named 'tmo_monitor.gateway'
```

hugoh commented:
Closing in favor of #49 .
